pkg/cmd/checkendpoints/cmd.go

```diff
@@ -60,10 +65,39 @@ func NewCheckEndpointsCommand() *cobra.Command {
 			kubeInformers.Core().V1().Secrets(),
 			recorder,
 		)
+
+		timeToStart := newTimeToStartController(
+			apiextensionsInformers.Apiextensions().V1().CustomResourceDefinitions(),
+			recorder,
+		)
+
+		stopController := newStopController(
+			apiextensionsInformers.Apiextensions().V1().CustomResourceDefinitions(),
+			recorder,
+		)
+
 		controller.RegisterMetrics()
+
+		// block until the PodNetworkConnectivityCheck CRD has been established
+		apiextensionsInformers.Start(ctx.Done())
+		ttsContext, ttsCancel := context.WithCancel(ctx)
+		go timeToStart.Run(ttsContext, 1)
```

Comment:
this feels super overengineered. Why not a simple wait.Until func that gets the CRD every 30 sec and blocks?

Reply:
I was under the impression that, overall, using Informer/Lister was a better practice than polling w/ get. The stop controller needed the informer, and our factory.Controller framework made it easy also use the informer in a "start controller" with minimal effort.